openid-4-verifiable-presentations-1_0.md

```diff
@@ -559,11 +565,276 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+
+`path`:
+: REQUIRED if the Credential Format uses a JSON-based claims structure; MUST NOT
+be present otherwise. The value MUST be a claims path pointer that specifies the path to the claim
+within the Verifiable Credential, as defined in (#claims_path_pointer).
+
+`namespace`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the namespace of the claim
+within the Verifiable Credential, e.g., `org.iso.18013.5.1`.
+
+`claim_name`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the name of the claim within the provided namespace
+in the Verifiable Credential, e.g., `first_name`.
+
+`value`:
+: OPTIONAL. A string, integer or boolean value that specifies the expected value of the claim. If the
+`value` property is present, the Wallet MUST return the claim only if the type and value
+of the claim match the type and value specified in the query.
+
+`values`:
+: OPTIONAL. An array of strings, integers or boolean values that specifies the expected values of the claim.
+If the `values` property is present, the Wallet MUST return the claim only if the
+type and value of the claim both match for at least one of the elements in the array.
+
+### Selecting Claims and Credentials {#vp_query_lang_processing_rules}
+
+The same basic logic applies for selecting claims and for selecting credentials, as detailed in the following.
+
+#### Selecting Claims
+
+The following rules apply for selecting claims via `expect_claims` and `valid_claim_sets`:
+
+- If `expect_claims` is not provided, the Verifier requests all claims existing
+  in the Credential.
+- If `expect_claims` is provided, but `valid_claim_sets` is not provided,
+  the Verifier requests all claims listed in `expect_claims`.
+- Otherwise, the Verifier requests one combination of the claims listed in
+  `valid_claim_sets`, with optional claims marked by the postfix `?`.
+
+If the Wallet cannot deliver all non-optional claims requested by the Verifier according to these rules, it MUST NOT
+return the respective Credential.
+
+#### Selecting Credentials
+
+The following rules apply for selecting Credentials via `expect_credentials` and `valid_credential_sets`:
+
+- If `valid_credential_sets` is not provided, the Verifier requests all
+  Credentials in `expect_credentials` to be returned.
+- Otherwise, the Verifier requests one combination of the Credentials
+  listed in `valid_credential_sets`, with optional credentials marked by the postfix `?`.
+
+Credentials not matching the respective constraints expressed within
+`expect_credentials` MUST NOT be returned, i.e., they are treated as if
+they would not exist in the Wallet.
+
+If the Wallet cannot deliver all non-optional Credentials requested by the Verifier according to these rules, it MUST NOT
+return any credential.
```

Comment:
```suggestion
If the Wallet cannot deliver all non-optional Credentials requested by the Verifier according to these rules, it MUST NOT
return any credential(s).
```

As some requests might involve the need to return multiple credentials and this sentence implied to me that only one could be returned.